src/annotator/plugin/bucket-bar-js.js

```diff
@@ -0,0 +1,52 @@
+import { getBoundingClientRect } from '../highlighter';
+
+// FIXME: Temporary duplication of size constants between here and BucketBar
+const BUCKET_SIZE = 16; // Regular bucket size
+const BUCKET_NAV_SIZE = BUCKET_SIZE + 6; // Bucket plus arrow (up/down)
+const BUCKET_TOP_THRESHOLD = 115 + BUCKET_NAV_SIZE; // Toolbar
+
+/**
+ * Find the closest valid anchor in `anchors` that is offscreen in the direction
+ * indicated.
+ *
+ * @param {Object[]} anchors
```

Comment:
I think it would be good to introduce a typedef for the anchor here, or at least the parts that the code in this file actually uses. To start with it could be as simple as:

```js
/**
 * @typedef Anchor
 * @prop {HTMLElement[]} [highlights]
 */
```

Reply:
@robertknight  Is it OK for the `anchor` typedef to live in this module for the moment, or...where should it live?